tutorials/compiler-explorer-deployment.md

```diff
@@ -0,0 +1,219 @@
+<!--
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+# Tutorial: Compiler Explorer Deployment
+
+Full instructins are available at [compiler-explorer: Adding a new library](https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingALibrary.md). This page is a Beman cheatsheet.
+
+## Integrating a header-only library
+
+### Step by step example: `beman.optional`
+
+1. Clone Compiler Explorer repos: <https://github.com/compiler-explorer/infra> and <https://github.com/compiler-explorer/compiler-explorer>.
+2. Update `infra/`:
+
+   2.1. Locate an existing Beman library inside `bin/yaml/libraries.yaml` - e.g. search for `repo: bemanproject/optional`.
+
+   2.2. Add a new entry (keeping Beman entries grouped and sorted):
+
+```yml
+beman_optional:
+  build_type: none
+  check_file: README.md
+  method: nightlyclone
+  repo: bemanproject/optional
+  targets:
+  - main
+  type: github
+```
+
+Notes:
+
+* `build_type: none`: Not required to run a build command (header-only libraries).
+* `check_file: README.md`: The platform checks the hash of this file.
+* `method: nightlyclone`: Do a deployment every day. We don't want and don't need more often deployment.
+* `repo: bemanproject/optional`: Clone and deploy `beman.optional` from <https://github.com/bemanproject/optional>.
+* `target: ... main`: Only deploy main as trunk version.
+* `type: github`: The repo will be always clone with latest version.
+
+  2.3. Deploy local `infra/` instance. For header-only libraries, it's enough to just run:
+
+```bash
+# Create default deployment directory and make sure to set write permissions for your user.
+$ mkdir -p /opt/compiler-explorer
+$ ls -l /opt/                  
+drwxr-xr-x  25 dariusn  wheel   800 Apr  7 21:29 compiler-explorer
+
+$ compiler-infra [main] make ce
```

Comment:
```suggestion 
$ make ce
```

`complier-infra [main]` seem like extra information that we don't need.

Reply:
done